redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/MemoryStateStore.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse
+
+/** A simple in-memory state store. */
+public class MemoryStateStore : StateStore {
+  private val map = mutableMapOf<String, StateSnapshot>()
+
+  override suspend fun put(id: String, value: StateSnapshot) {
```

Comment:
I think we probably need to document the concurrency expectations for this. Perhaps as simple as ‘all functions are invoked on the UI dispatcher’

Reply:
(on the class-level doc actually)